### PR TITLE
Add 'Admin' view to new competitions table

### PR DESCRIPTION
**Pull request by gregorbg**
*CHANGE* Previously, "Admin" used to be a separate view option next to "List" and "Map". But it's actually also a list so I decided to just turn it into a checkbox.

It is only shown when you are logged in as somebody who actually has admin privileges.
image

Clicking the checkbox starts rendering additional admin details like on the old page
image

### CODE REVIEW

File: app/webpacker/components/CompetitionsOverview/CompetitionsFilters.js [hunk 1]
```diff
@@ -52,12 +54,22 @@ function CompetitionsFilters({
         </Form.Field>
       </Form.Group>
 
+      {canViewAdminDetails && shouldShowAdminDetails && (
+        <Form.Group>
+          <Form.Field>
+            <AdminStatusButtonGroup filterState={filterState} dispatchFilter={dispatchFilter} />
+          </Form.Field>
+        </Form.Group>
+      )}
```
Comment (kr-matthews) [inline app/webpacker/components/CompetitionsOverview/CompetitionsFilters.js:63]: Can we put this below the checkboxes? It feels weird when clicking a checkbox moves that checkbox down, so you can't directly click again to un-check it.

(Just my initial thoughts from running locally; will review the code later.)

File: app/webpacker/components/CompetitionsOverview/CompetitionsView.js [hunk 2]
```diff
@@ -38,8 +38,14 @@ function CompetitionsView() {
...
-    [debouncedFilterState],
+    () => calculateQueryKey(debouncedFilterState, canViewAdminDetails),
+    [debouncedFilterState, canViewAdminDetails],
+  );
+
+  // Need to make sure that people don't "hijack" admin mode by manipulating the URL
+  const shouldShowAdminDetails = useMemo(
+    () => canViewAdminDetails && filterState.shouldShowAdminDetails,
+    [canViewAdminDetails, filterState.shouldShowAdminDetails],
   );
```
Comment (kr-matthews) [inline app/webpacker/components/CompetitionsOverview/CompetitionsView.js:49]: There's no benefit to memoizing a simple calculation resulting in a boolean.
```suggestion
  const shouldShowAdminDetails = canViewAdminDetails && filterState.shouldShowAdminDetails;
```

File: app/webpacker/components/CompetitionsOverview/queryUtils.js [hunk 3]
```diff
@@ -4,26 +4,32 @@ import { competitionConstants } from '../../lib/wca-data.js.erb';
...
   if (filterState?.timeOrder === 'past') {
     timeKey = `${filterState.selectedYear}`;
   } else if (filterState?.timeOrder === 'custom') {
     timeKey = `start${filterState.customStartDate}-end${filterState.customEndDate}`;
   }
 
+  let adminStatus = filterState?.adminStatus;
+  if (!canViewAdminDetails) {
+    adminStatus = null;
+  }
```
Comment (kr-matthews) [inline app/webpacker/components/CompetitionsOverview/queryUtils.js:18]:
```suggestion
  const adminStatus = canViewAdminDetails ? filterState?.adminStatus : null
```

**Review by kr-matthews — COMMENTED** [review]
A few files are getting a bit long, it might be good to break some of them up.

**Comment by gregorbg** [comment]
I want to leave code cleanup for when the feature has been sufficiently tested and is about to be launched

**Pull request merged**